core/src/main/java/org/infinispan/interceptors/locking/ClusteringDependentLogic.java

```diff
@@ -349,6 +360,12 @@ protected void commitSingleEntry(CacheEntry entry, Metadata metadata, FlagAffect
          // TODO: Can the reset be done after notification instead?
          boolean created = entry.isCreated();
          boolean removed = entry.isRemoved();
+         boolean expired;
+         if (entry instanceof MVCCEntry) {
```

Comment:
Also I changed it to just ``if (removed && entry instanceof MVCCEntry)``  Since remove expiried sets both removed and expired.